### PR TITLE
feat!: use ISender and deprecate Light Push `.push`

### PR DESCRIPTION
**Pull request by weboko**
## Problem

We don't have dedicated interface for Sending messages though we have `LightPush` and `Relay` that do similar thing.

## Solution

Define and implement `ISender` interface.

## Notes

- Related to https://github.com/waku-org/js-noise/issues/14

### CODE REVIEW

File: packages/interfaces/src/light_push.ts [hunk 1]
```diff
@@ -4,11 +4,13 @@ import type {
...
-export interface ILightPush extends PointToPointProtocol {
-  push: (
-    encoder: IEncoder,
-    message: IMessage,
-    opts?: ProtocolOptions
-  ) => Promise<SendResult>;
-}
+export type ILightPush = ISender &
+  PointToPointProtocol & {
+    push: (
```
Comment (weboko) [inline packages/interfaces/src/light_push.ts:11]: I think it should be simplified to `sender`
@fryorcraken any reasons not to?

Comment (fryorcraken) [reply packages/interfaces/src/light_push.ts:11]: I agree that part of the sender/receiver work, it'll be a good opportunity to modify the existing API so they can all have the same function signature (ie, lightpush push and relay send, filter subscribe and relay add observer)

Comment (weboko) [reply packages/interfaces/src/light_push.ts]: Renamed to `send`.

File: packages/core/src/lib/light_push/index.ts [hunk 2]
```diff
@@ -40,14 +40,6 @@ class LightPush extends BaseProtocol implements ILightPush {
     encoder: IEncoder,
     message: IMessage,
     opts?: ProtocolOptions
-  ): Promise<SendResult> {
-    return this.push(encoder, message, opts);
-  }
-
-  async push(
```
Comment (fryorcraken) [inline packages/core/src/lib/light_push/index.ts:39]: I would mark this as `@deprecated` for now so make migration easier.

File: packages/core/CHANGELOG.md
Comment (fryorcraken) [inline packages/core/CHANGELOG.md:1]: I guess this is not meant to be in this PR, is it?

**Review by fryorcraken — COMMENTED** [review]
The PR is not clean (see CHANGELOG files)

File: packages/core/CHANGELOG.md
Comment (weboko) [reply packages/core/CHANGELOG.md:1]: Right, accidentally pushed

**Review by fryorcraken — APPROVED** [review]

**Pull request merged**